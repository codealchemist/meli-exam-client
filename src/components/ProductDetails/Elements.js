import styled from 'styled-components'
import Colors from '../Colors'

export const Wrapper = styled.div`
  width: 100%;
  background: ${Colors.default.light1};
  padding: 20px;
`

export const TopWrapper = styled.div`
  display: flex;
  padding: 20px;
`

export const ImageWrapper = styled.div`
  flex: 1;
  text-align: center;
`

export const BottomWrapper = styled.div`
  width: 100%;
`

export const BuyWrapper = styled.div`
  width: 300px;
  text-transform: capitalize;

  button {
    background: ${Colors.default.primary};
    font-size: 15px;
    padding: 12px 16px;
    border: 1px solid ${Colors.default.primary};
    color: ${Colors.default.light1};
    cursor: pointer;
    text-align: center;
    border-radius: 4px;
  }
`
