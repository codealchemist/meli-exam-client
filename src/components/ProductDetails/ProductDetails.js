import React from 'react'
import CurrencyFormat from 'react-currency-format'
import Context from '../../Context'
import {
  Wrapper,
  TopWrapper,
  ImageWrapper,
  BuyWrapper,
  BottomWrapper,
  Image,
  Content
} from './Elements'

const ProductDetails = () => (
  <Wrapper>
    <Context.Consumer>
      {context => (
        <>
          <TopWrapper>
            <ImageWrapper>
              <Image src={context.productDetails.item.picture} />
            </ImageWrapper>
            <BuyWrapper>
              <label>
                {context.productDetails.item.condition} -{' '}
                {context.productDetails.item.sold_quantity} vendidos
              </label>
              <h2>{context.productDetails.item.title}</h2>
              <h1>
                <CurrencyFormat
                  displayType="text"
                  prefix={'$'}
                  thousandSeparator="."
                  decimalSeparator=","
                  value={context.productDetails.item.price.amount}
                />
              </h1>
              <button>Comprar</button>
            </BuyWrapper>
          </TopWrapper>
          <BottomWrapper>
            <Content>
              <h3>Descripci√≥n del producto</h3>
              <pre>{context.productDetails.item.description}</pre>
            </Content>
          </BottomWrapper>
        </>
      )}
    </Context.Consumer>
  </Wrapper>
)

export default ProductDetails
