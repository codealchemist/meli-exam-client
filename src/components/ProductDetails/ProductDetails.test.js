import React from 'react'
import { mount } from 'enzyme'
import ProductDetails from './ProductDetails'

jest.mock('../../Context', () => ({
  Consumer: obj =>
    obj.children({
      productDetails: {
        item: {
          picture: 'image-url',
          condition: 'new',
          sold_quantity: 86,
          title: 'Rock',
          price: {
            amount: 42
          },
          description: 'Description.'
        }
      }
    })
}))

describe('ProductDetails', () => {
  it('should render', () => {
    const wrapper = mount(<ProductDetails />)
    expect(wrapper).toMatchSnapshot()
  })
})
