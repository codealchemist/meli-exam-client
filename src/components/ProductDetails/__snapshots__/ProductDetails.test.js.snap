// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails should render 1`] = `
<ProductDetails>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "hwUqSB",
            "rules": Array [
              "
  width: 100%;
  background: ",
              "#fff",
              ";
  padding: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa hwUqSB"
      >
        <Consumer>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "lastClassName": "ducbCl",
                    "rules": Array [
                      "
  display: flex;
  padding: 20px;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bwzfXH ducbCl"
              >
                <styled.div>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": true,
                          "lastClassName": "jeapRr",
                          "rules": Array [
                            "
  flex: 1;
  text-align: center;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-htpNat jeapRr"
                    >
                      <img
                        alt="Product"
                        src="image-url"
                      />
                    </div>
                  </StyledComponent>
                </styled.div>
                <styled.div>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-ifAKCX",
                          "isStatic": true,
                          "lastClassName": "cYCRKM",
                          "rules": Array [
                            "
  width: 300px;
  text-transform: capitalize;

  button {
    background: ",
                            "#3483fa",
                            ";
    font-size: 15px;
    padding: 12px 16px;
    border: 1px solid ",
                            "#3483fa",
                            ";
    color: ",
                            "#fff",
                            ";
    cursor: pointer;
    text-align: center;
    border-radius: 4px;
  }
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-ifAKCX",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-ifAKCX cYCRKM"
                    >
                      <label>
                        new
                         -
                         
                        86
                         vendidos
                      </label>
                      <h2>
                        Rock
                      </h2>
                      <h1>
                        $ 
                        42
                      </h1>
                      <button>
                        Comprar
                      </button>
                    </div>
                  </StyledComponent>
                </styled.div>
              </div>
            </StyledComponent>
          </styled.div>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bxivhb",
                    "isStatic": true,
                    "lastClassName": "jcNPuI",
                    "rules": Array [
                      "
  width: 100%;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bxivhb",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bxivhb jcNPuI"
              >
                <h3>
                  Descripci√≥n del producto
                </h3>
                <p>
                  Description.
                </p>
              </div>
            </StyledComponent>
          </styled.div>
        </Consumer>
      </div>
    </StyledComponent>
  </styled.div>
</ProductDetails>
`;
